src/mappings/swaps/HydraDx.ts

```diff
@@ -75,6 +116,20 @@ export function findHydraDxFee(events: EventRecord[]): Fee | undefined {
     }
 }
 
+function isPartOfRouterSwap(events: TypedEventRecord<Codec[]>[]): boolean {
+    const eventRecords = events as EventRecord[]
+    for (const eventRecord of eventRecords) {
+        if (
+            eventRecord.event.section == "router" && 
+            eventRecord.event.method == "RouteExecuted"
+        ) {
+            return true
+        }
+    }
+
+    return false
```

Comment:
We can also use `events.any((event) =>  event.event.section == "router" && 
            event.event.method == "RouteExecuted")` iirc